arangod/RestHandler/RestAdminExecuteHandler.cpp

```diff
@@ -61,9 +61,16 @@ RestStatus RestAdminExecuteHandler::execute() {
 
   TRI_ASSERT(server().getFeature<V8DealerFeature>().allowAdminExecute());
 
-  std::string_view bodyStr = _request->rawPayload();
-  char const* body = bodyStr.data();
-  size_t bodySize = bodyStr.size();
+  std::string_view payload;
+  size_t bodySize;
+  if (_request->header(StaticStrings::ContentTypeHeader) == "text/plain") {
```

Comment:
I don't think this is sufficient. Though it will work if the content-type is exactly `text/plain`, it won't work if the content-type is `text/plain` plus any charset (see https://www.w3.org/International/articles/http-charset/index.en). This should also be supported and work.